R/class.R

```diff
@@ -0,0 +1,76 @@
+
+#' @title Numbat R6 class
+#' @description 
+#' @param out_dir the Numbat run output directory
+#' @export
+Numbat <- R6::R6Class("Numbat", lock_objects=FALSE,
+  public = list(
+    #' @field label
+    label = 'sample',
+
+    initialize = function(out_dir, verbose=TRUE) {
+
+        self$out_dir = out_dir
+
+        self$fetch_results(out_dir)
+
+    },
+
+    fetch_results = function(out_dir, i = 2) {
```

Comment:
I would put this in the initialize step. 

Users must initialize the Numbat object with these files. When these files are loaded, that's a Numbat object. 

```
num1 = Numbat(joint_point = ...)
```
Check how an input is required for pagoda2: https://github.com/kharchenkolab/pagoda2/blob/main/R/Pagoda2.R#L79-L95

and Conos

Reply:
Update:

Discussed on slack. @teng-gao 

--- Move the logic of `fetch_results()` into the constructor (i.e. `initialize()`
--- Check for missing files. If missing, throw fatal error
--- Probably should add simple checks to make sure that the files are valid